### PR TITLE
8284903: Fix typos in hotspot

### PR DESCRIPTION
**Pull request by magicus**
I ran `codespell` on hotspot, and accepted those changes where it indeed discovered real typos. 

You'd be surprised over the many implementions of instrinsics and other intructions accross all archtectures I've encounted, so for the preceding reason it's neccesery to sucessfully seach for exisiting typos...

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8284903](https://bugs.openjdk.java.net/browse/JDK-8284903): Fix typos in hotspot

### Reviewers
 * [Chris Plummer](https://openjdk.java.net/census#cjplummer) (@plummercj - **Reviewer**) ⚠️ Review applies to 449d8a2395a31b96261419410734d45c30339ab4
 * [Coleen Phillimore](https://openjdk.java.net/census#coleenp) (@coleenp - **Reviewer**) ⚠️ Review applies to 449d8a2395a31b96261419410734d45c30339ab4
 * [Vladimir Kozlov](https://openjdk.java.net/census#kvn) (@vnkozlov - **Reviewer**) ⚠️ Review applies to 449d8a2395a31b96261419410734d45c30339ab4
 * [Lutz Schmidt](https://openjdk.java.net/census#lucy) (@RealLucy - **Reviewer**) ⚠️ Review applies to 449d8a2395a31b96261419410734d45c30339ab4
 * [Stefan Karlsson](https://openjdk.java.net/census#stefank) (@stefank - **Reviewer**) ⚠️ Review applies to 449d8a2395a31b96261419410734d45c30339ab4

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/8260/head:pull/8260` \
`$ git checkout pull/8260`

Update a local copy of the PR: \
`$ git checkout pull/8260` \
`$ git pull https://git.openjdk.java.net/jdk pull/8260/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 8260`

View PR using the GUI difftool: \
`$ git pr show -t 8260`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/8260.diff">https://git.openjdk.java.net/jdk/pull/8260.diff</a>

</details>

### CODE REVIEW

File: src/hotspot/share/prims/jvmtiThreadState.hpp [hunk 1]
```diff
@@ -133,7 +133,7 @@ class JvmtiThreadState : public CHeapObj<mtInternal> {
  public:
   ~JvmtiThreadState();
 
-  // is event_type enabled and usable for this thread in any enviroments?
+  // is event_type enabled and usable for this thread in any environments?
```
Comment (plummercj) [inline src/hotspot/share/prims/jvmtiThreadState.hpp:136]: Should it be "environment"?

File: src/hotspot/share/services/memTracker.hpp [hunk 2]
```diff
@@ -125,7 +125,7 @@ class MemTracker : AllStatic {
   }
 
   // Record arena size change. Arena size is the size of all arena
-  // chuncks that backing up the arena.
+  // chunks that backing up the arena.
```
Comment (plummercj) [inline src/hotspot/share/services/memTracker.hpp:128]: This doesn't read correctly (not due to your change). Perhaps "that back up" or "that are backing up" is what was intended.

**Review by plummercj — APPROVED** [review]
I tried to cover all serviceability files, but didn't want to pick and choose individual files within any given directory, so I reviewed all files in any directory that contained a serviceability related file. So the following were covered by my review:

src/os
src/os_cpu
src/share/prims
src/share/services

**Review by coleenp — APPROVED** [review]
I checked all directories except gc, jfr, metaspace, opto and jvmti.

**Review by vnkozlov — APPROVED** [review]
Nice work!

I reviewed compiler related code:
`cpu`, `os`, `os_cpu`, `share/adlc`, `share/c1`, `share/ci`, `share/code`, `share/compiler`, `share/jvmci`, `share/opto`.

File: src/hotspot/share/runtime/os.cpp [hunk 3]
```diff
@@ -641,7 +641,7 @@ void* os::malloc(size_t size, MEMFLAGS memflags, const NativeCallStack& stack) {
 
   DEBUG_ONLY(check_crash_protection());
 
-  // On malloc(0), implementators of malloc(3) have the choice to return either
+  // On malloc(0), implementers of malloc(3) have the choice to return either
```
Comment (dholmes-ora) [inline src/hotspot/share/runtime/os.cpp:644]: I suggest "implementations of malloc..."

File: src/hotspot/share/runtime/task.hpp [hunk 4]
```diff
@@ -69,7 +69,7 @@ class PeriodicTask: public CHeapObj<mtInternal> {
   // between T and T + interval_time.
   void enroll();
 
-  // Make the task deactive
+  // Make the task deactivate
```
Comment (dholmes-ora) [inline src/hotspot/share/runtime/task.hpp:72]: Should be "inactive"

File: src/hotspot/share/services/threadIdTable.cpp [hunk 5]
```diff
@@ -92,7 +92,7 @@ void ThreadIdTable::lazy_initialize(const ThreadsList *threads) {
   if (!_is_initialized) {
     {
       // There is no obvious benefits in allowing the thread table
```
Comment (dholmes-ora) [inline src/hotspot/share/services/threadIdTable.cpp:94]: s/benefits/benefit/

File: src/hotspot/share/services/threadIdTable.cpp [hunk 6]
```diff
@@ -92,7 +92,7 @@ void ThreadIdTable::lazy_initialize(const ThreadsList *threads) {
...
-      // to be concurently populated during the initalization.
+      // to be concurrently populated during the initialization.
```
Comment (dholmes-ora) [inline src/hotspot/share/services/threadIdTable.cpp:95]: delete "the"

**Review by dholmes-ora — CHANGES_REQUESTED** [review]
Hi Magnus,

Generally looks good. Thanks for doing this extremely tedious cleanup.

A few comments/suggestions/corrections below.

And of course copyright dates need updating for Oracle copyrights.

Thanks,
David

**Comment by dougxc** [comment]
Are there plans to add `codespell` to the make system to somehow catch/prevent spelling regressions in future? I understand this may not be feasible.

**Comment by XenoAmess** [comment]
> Are there plans to add `codespell` to the make system to somehow catch/prevent spelling regressions in future? I understand this may not be feasible.

IMO all spell checking mechanisms suffers from false positive and need human manual job to filter them...
So maybe we can only arrange a scan per year or per months...

**Comment by magicus** [comment]
Thank you for all your input! I will update the PR according to all suggestions. But let me answer a few more general comments first.

The spelling corrections in this PR was generated by codespell. I manually removed all false positives (that was the most tedious part of this fix). Codespell works by checking a list of commonly misspelled words, not by verifying that all words are present in a list of accepted spellings. The latter approach might work for normal prose, but is utterly worthless for source code. In practice, codespell does automatically what several users have been doing manually lately -- discover a common misspelling, searching the code base for more instances of that misspelling. So, there are certainly lots of typos left. But this should reasonably cut down on the number of PRs of the type "Fix typos for supercalifragilisticexpialidocious" that updates a lot of files for just a single misspelled word.

For some cases where I were not sure (like mismatch or re-enable), I googled, read spelling recommendations and/or checked how common alternative spellings where. When the ratio exceeded 10:1 or so, I deemed one alternative as "correct"; in less common cases I let the original authors choice of spelling remain.

I have not read the complete sentences, just verified that the actual word fixes seemed to make sense. Hence the odd results in some places. 

Ideally, I would like to find some way of keeping the code "codespell clean" using tooling. Due to the amount of false positives, this is not a trivial task. I don't think we'll ever be able to have it as part of jcheck, but we might have a script that someone runs once per release or so. I'm not yet sure how to achieve this, but my two main ideas is to either just run codespell on added or modified code (think "+" lines in git diff), or to have a list of acknowledged false positives that should be skipped.

Before I can even start on this, the huge number of *true* positives needs to be addressed, though! Hotspot is by no means worst, it's just a huge chunk of code which I addressed all at once.

**Review by RealLucy — APPROVED** [review]
This is a partial approval for 
 - cpu/ppc
 - cpu/s390
 - os/aix
 - os_cpu/aix_ppc
 - os_cpu/linux_ppc
 - os_cpu/linux_s390

Thank you for completing this tedious job!

**Review by stefank — APPROVED** [review]
Reviewed:
gc/
oops/
utilities/

File: src/hotspot/share/runtime/task.hpp [hunk 7]
```diff
@@ -69,7 +69,7 @@ class PeriodicTask: public CHeapObj<mtInternal> {
   // between T and T + interval_time.
   void enroll();
 
-  // Make the task deactivate
+  // Make the task inactivate
```
Comment (dholmes-ora) [inline src/hotspot/share/runtime/task.hpp:72]: That should have been "inactive".

**Pull request closed without merging**